stubs/google/cloud/ndb/query.pyi

```diff
@@ -0,0 +1,102 @@
+from . import Cursor, Key, TYPE_MODEL, Property
+from typing import (
+    Any, Iterator, List, Optional, Sequence, TypeVar, Tuple, Union,
+    overload)
+from typing_extensions import Literal
+
+_T = TypeVar('_T', covariant=True)
+
+class QueryOptions: ...
+
+class PropertyOrder:
+    name: str
+    reverse: bool
+    def __neg__(self) -> PropertyOrder: ...
+
+class RepeatedStructuredPropertyPredicate: ...
+
+class ParameterizedThing: ...
+
+class Parameter(ParameterizedThing): ...
+
+class ParameterizedFunction(ParameterizedThing): ...
+
+class Node: ...
+
+class FalseNode(Node): ...
+
+class ParameterNode(Node):
+    _prop: str
+    _op: str
+    _param: str
+
+class FilterNode(Node):
+    _name: str
+    _opsymbol: str
+    _value: str
+
+class ConjunctionNode(Node):
+    def __iter__(self) -> Iterator[Node]: ...
+
+class DisjunctionNode(Node): ...
+
+AND = ConjunctionNode
+OR = DisjunctionNode
+
+class Query:
```

Comment:
Why did you remove the generics?

Reply:
You will not need the added types in some files then.

Reply:
Because the real class is not generic. I feel adding generics to this will just make our jobs harder in the future, so I've removed them.

Reply:
Hmm, I think in our use case so far `Query` returns only one type of model, I don't see the problem with this approach. Should we keep it until we need to get rid of it?

Reply:
I think, since I've already done the work to get rid of it, it's quicker to just do now. If we really want to save on manual type annotations, we can just introduce helpers to `BaseModel` that returns the correct type later.

Reply:
Okay, fine with that, we got rid of casts (that were there before generics) so the most important part is resolved.